pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -557,6 +557,30 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, nil
 	}
 
+	// Ensure that if DNSZone already exists, it's PreserveOnDelete setting matches the ClusterDeployment:
+	if cd.Spec.ManageDNS {
+		dnsZone := &hivev1.DNSZone{}
+		nsName := types.NamespacedName{Namespace: cd.Namespace, Name: controllerutils.DNSZoneName(cd.Name)}
+		logger := cdLog.WithField("zone", nsName.String())
+
+		switch err := r.Get(context.TODO(), nsName, dnsZone); {
+		case err != nil && !apierrors.IsNotFound(err):
```

Comment:
IIUC, the first pass through for a *fresh* cluster would now hit here with `IsNotFound` and fall out of this whole chunk.

Then further down we `ensureManagedDNSZone` which creates the DNSZone.

That triggers another Reconcile here, because we watch DNSZones and enqueue for the owning CD.

And it's on that _second_ reconcile we would hit this new code path to sync up the `PreserveOnDelete` settings.

Reply:
I didn't think it would trigger, if it is an IsNotFound error, we don't have a case for that in the switch, which I think would be a no-op and continue with the code normally to create the DNSZone, which will get the flag on creation. Does that seem right?

Reply:
> which will get the flag on creation

Ahh, I see that now. Cool.